src/niftyone/figures/generator.py

```diff
@@ -92,10 +94,23 @@ def __call__(
         overwrite: bool,
     ) -> None:
         # Filters by entity (via string query)
-        table_idxes = table.ent.query(self.query).index
-        for idx in table_idxes:
+
+        # First query is for main image, subsequent are for overlays
+        main_idxes = table.ent.query(self.queries[0]).index
+        if overlays := len(self.queries) > 1:
+            overlay_idxes = table.ent.query(
```

Comment:
Making a note of something we discussed offline. I think it could be nicer to handle this more generically using a join strategy. Something like this perhaps

```python
query_dfs = [table.ent.query(q) for q in self.queries]
indexed_dfs = [
    (
        df
        .reset_index()
        .loc[:, ["index"] + self.join_entities]
        .rename(columns={"index": f"index_{ii}"})
    )
    for ii, df in enumerate(query_dfs)
]
joined = indexed_dfs[0].join(indexed_dfs[1:], on=self.join_entities)
indices = [joined[f"index_{ii}"] for ii in range(len(self.queries))]

for inds in zip(indices):
    records = [df.loc[ind] for ind, df in zip(inds, query_dfs)]
    self.generate(*records)
```

Reply:
Spoke about this offline, just making some notes:

* The joined dataframe should still be able to match in the case of multiple records in main (re: first) query
* Expose join entity in config with a sensible default

Reply:
Updated in [37bfbbf](https://github.com/childmindresearch/niftyone/pull/21/commits/37bfbbf50ae9f90ee46dc430e2b54f1487184ed5). I've still limited the number of overlays to 1 in this commit. Will handle multi-overlays in another PR down the road.